iroh-embed/src/p2p.rs

```diff
@@ -0,0 +1,128 @@
+//! The p2p service to use in an iroh system.
+
+use std::path::PathBuf;
+
+use anyhow::Result;
+use iroh_one::mem_p2p;
+use iroh_p2p::{Config as P2pConfig, Libp2pConfig};
+use iroh_rpc_types::p2p::P2pAddr;
+use iroh_rpc_types::store::StoreAddr;
+use iroh_rpc_types::Addr;
+use tokio::task::JoinHandle;
+
+// TODO:
+//
+// - Need to allow configuring in memory keystore
+// - make Lib2p2Config non_exhaustive and provide a builder
+
+/// The iroh peer-to-peer (p2p) service.
```

Comment:
general thought: I wonder if we should call this `networking` rather than `p2p` (including `iroh-p2p`)

Reply:
Hum, I'm not convinced `networking` is a good name - it is very general.  I don't have anything against the current naming, it helps you understand what it does.

Reply:
fair, just a thought